File: 2023/tests/day17.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day17');

const example1 = `
2413432311323
3215453535623
3255245654254
3446585845452
4546657867536
1438598798454
4457876987766
3637877979653
4654967986887
4564679986453
1224686865563
2546548887735
4322674655533
`.trim();
const example2 = `
111111111111
999999999991
999999999991
999999999991
999999999991
`.trim();

const input = fs.readFileSync('inputs/input17.txt', 'utf8');

describe('part 1', () => {
  test('example 1', () => {
    expect(part1(example1)).toBe(102);
  });

  test('solution', () => {
    expect(part1(input)).toBe(870);
  });
});

describe('part 2', () => {
  test('example 1', () => {
    expect(part2(example1)).toBe(94);
  });

  test('example 2', () => {
    expect(part2(example2)).toBe(71);
  });

  test('solution', () => {
    expect(part2(input)).toBe(1063);
  });
});

File: 2024/AdventOfCode/Tests/Day17Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day17Test {
    private const string Example = "Inputs/17-Example.txt";
    private const string Example2 = "Inputs/17-Example2.txt";
    private const string Input = "Inputs/17.txt";
    private static Day17 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day17(Example);
            const string expected = "4,6,3,5,6,3,5,2,1,0";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day17(Input);
            const string expected = "2,1,0,1,7,2,5,0,3";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample2() {
            _day = new Day17(Example2);
            const string expected = "117440";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day17(Input);
            const string expected = "267265166222235";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}